content/kb/troubleshooting-create-support-bundle-with-curl.md

```diff
@@ -0,0 +1,57 @@
+---
+title: "Troubleshooting: Create Support Bundle with cURL"
+author: Chin-Ya Huang
+draft: false
+date: 2021-04-12
+categories:
+  - "support bundle"
+---
+
+## Applicable versions
+
+All Longhorn versions.
+
+## Symptoms
+
+Not able to create the support bundle with a web browser.
+
+## Solution
+
+1. Expose Longhorn backend service. Below is an example using `NodePort`, you can also export via load balancer if one is set up.
+    ```
+    ip-172-30-0-21:~ # kubectl -n longhorn-system     patch svc longhorn-backend -p '{"spec":    {"type":"NodePort"}}'
+    service/longhorn-backend patched
+    ip-172-30-0-21:~ # kubectl -n longhorn-system get     svc/longhorn-backend
+    NAME               TYPE       CLUSTER-IP          EXTERNAL-IP   PORT(S)          AGE
+    longhorn-backend   NodePort   10.43.136.157       <none>        9500:32595/TCP   156m
+    ```
+
+2. Run the below script to create and download the support bundle.
+You will need to replace the `BACKEND_URL`, `ISSUE_URL`, `ISSUE_DESCRIPTION`.
+    ```
+    # Replace this block ====>
+    BACKEND_URL="18.141.237.97:32595"
```

Comment:
Can also have script way of getting the `BACKEND_URL`?

NODEPORT=$(kubectl get services longhorn-backend -n longhorn-system -o jsonpath="{.spec.ports[0].nodePort}" )
NODES=($(kubectl get nodes -o jsonpath='{ $.items[*].status.addresses[?(@.type=="InternalIP")].address }'))
BACKEND_URL="${NODES[0]}:${NODE_PORT}"

Reply:
This should be nice to have for a KB, but thanks for that suggestion :+1:

Reply:
I've left this one for the user to change because NodePort might not be used depending on the user cluster set up.